src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2981,7 +2981,8 @@ private BoundExpression BindIsOperator(BinaryExpressionSyntax node, DiagnosticBa
                     //    The value is always false.
 
                     Debug.Assert(targetType.IsNullableType());
-                    return (operandType == targetType.GetNullableUnderlyingType()) ? ConstantValue.True : ConstantValue.False;
```

Comment:
When we are checking if types are identity convertable from the CLR point of view, all aspects should be ignored.